src/iris/api.py

```diff
@@ -1465,9 +1538,14 @@ def on_get(self, req, resp):
         fields = [f for f in fields if f in plan_columns] if fields else None
         req.params.pop('fields', None)
         if not fields:
-            fields = plan_columns
+            fields = list(plan_columns.keys())
+        requested_fields = list(fields)
```

Comment:
we don't need to do that cast here since above it looks like fields will always be a list

Reply:
It was done so that requested_fields is a shallow copy of the original field names, so later modifications to fields (like removing or reordering items) won’t affect requested_fields